### ISSUE TITLE
Cannot compile

### ISSUE THREAD

**Issue by wjvanpatten** [connormeredithcu/Meredith_CSCI2270_FinalProject#1]
To whom it may concern,

_Warning, though I am doing very well in 2270 I still consider myself a novice programmer_

I have successfully cloned this repository onto my VM but **cannot compile the file.**

To prevent a spam-like email, I have attached a .txt file with the error output from CodeBlocks.

I think code blocks is having a hard time with determining dependencies and fixing this has proven to be over my head (I've been scouring stack exchange for about 30 minutes to solve the problem).

Any thoughts?

Thanks.

~John
[Meredith Unable to Compile.txt](https://github.com/connormeredithcu/Meredith_CSCI2270_FinalProject/files/243377/Meredith.Unable.to.Compile.txt)

**Comment by connormeredithcu** [comment]
Hi John,

Can you please tell me what compiler you are using and any compiler settings you have set? You can find these by selecting Project->Build Options... in Code Blocks. Once I get this info, I'll do my best to replicate your setup in the VM and fix what's going wrong. I actually programmed my final project outside of the VM, oh no! :(

Looking through the error output you attached, it seems to me like we're dealing with compiler incompatibilities. I think this because things like "undefined reference to `___gxx_personality_sj0'|" don't seem to present issues with the written code, but the way my project was built at first.

I would very much appreciate it if you were to try the following for me:
**1.** Create a new project and call it whatever you'd like.
**2.** Create a new class called, exactly, "ResponseHandler" without quotes and add it to the new project.
**3.** Create a new header file called "Utilities" without quotes in the "include" folder of the new project. Make sure to check any boxes about adding the header to the current project. (File->New...->File...->C/C++ Header)
**4.** Create a new source file called "Utilities" without quotes in the "src" folder of the new project. Once again, make sure to have this file added to the project. (File->New...->File...->C/C++ Source)
**5.** Copy the contents of the files in the old project to their twins in the new project. i.e.,
    -Copy the contents of my "main.cpp" to the new "main.cpp" in the new project.
    -Copy the contents of my "Utilities.cpp" to the new "Utilities.cpp" in the new project.
    -Copy the contents of my "ResponseHandler.cpp" to the new "ResponseHandler.cpp" in the new project.
    -Copy the contents of my "Utilities.h" to the new "Utilities.h" in the new project.
    -Copy the contents of my "ResponseHandler.h" to the new "ResponseHandler.h" in the new project.
**6.** Make sure the new project is active and try to build.
In summary, I'd just like to see what happens if we make a project with the identical code, but without my compiler mucking around with your compiler. No worries if you don't want to go through this process, but I hope it'll help if you do!

Thanks for your patience,
Connor

**Comment by connormeredithcu** [comment]
Hi again!

I tried cloning my project to the VM, and sure enough, I got the same set of errors as you. Going through the step by step process helped immensely, and then I fixed several Ubuntu specific issues. Please update your repository on your end (Using git pull origin master) and use the VMAlternateVersion project instead of the one made for Windows 8.1. Also, please make sure that under Project->Build Options... the "Have g++ follow the 1998 ISO C++ language standard" is selected.

Let me know how it works out!

Thanks again,
Connor

**Comment by wjvanpatten** [comment]
Thank you for the quick response. Will look into this tomorrow.

**Comment by connormeredithcu** [comment]
Great, thank you.

**Comment by trippygif** [comment]
Hey @connormeredithcu I am having the same trouble. I apparently do not have permission to run the file on the virtual machine. Is there any other way to execute the program outside of CodeBlocks?

**Comment by connormeredithcu** [comment]
Yes, you can definitely run it outside of CodeBlocks!

**If you want to stay in the VM**, then navigate to LocationOfRepo\Meredith_CSCI2270_FinalProject\VMAlternateVersion\bin\Debug in the terminal and run the command "./VMAlternateVersion" without quotes. If you get an error like "bash: ./VMAlternateVersion: Permission denied" then run the command "chmod u+x VMAlternateVersion" without quotes and again try the command "./VMAlternateVersion".

**If you want to use windows**, then clone the repository to your machine (Or just have it in your dropbox,) and navigate to the folder LocationOfRepo\Meredith_CSCI2270_FinalProject\Connor_Meredith_FinalProject_CSCI2270\bin\Debug and run the .exe file you find there. This is where CodeBlocks automatically creates an executable version of the program for future reference.

I'm sorry for all the issues. It was working great on my end! Please let me know how this works.

**Comment by trippygif** [comment]
@connormeredithcu I tried to run on Windows 10 but again access was denied

**Comment by connormeredithcu** [comment]
Hi again,

I'm sorry, but I can't do much about windows 10. I only know how it's working on Windows 8.1 64 bit and Ubuntu 64 bit (as said in the readme.) I don't know about how file permissions work on Windows 10.
Can you try the process for running the executable on the VM? The chmod command I mentioned will fix the permissions issue as far as I can tell.

Sorry again for all these issues. It was originally only written for windows 8.1 so I'm doing my best to accommodate these new needs :(

**Comment by trippygif** [comment]
Hey I got it to work!

**Comment by connormeredithcu** [comment]
Yayyyyyy! I'm glad to hear it.
